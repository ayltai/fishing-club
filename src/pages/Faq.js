// @flow
'use strict';

import React from 'react';
import muiThemeable from 'material-ui/styles/muiThemeable';

const Faq = () : any => {
    return (
        <div>FAQ</div>
    );
};

export default muiThemeable()(Faq);
