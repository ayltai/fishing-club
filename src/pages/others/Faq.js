// @flow
'use strict';

import React from 'react';
import muiThemeable from 'material-ui/styles/muiThemeable';
import FaqItem from './FaqItem';

class Faq extends React.Component {
    componentDidMount() : void {
        window.scrollTo(0, 0);
    }

    render() : any {
        return (
            <div>
                <FaqItem
                    question="Are we going to rent a yacht and fishing at sea?"
                    answer={<div>Not exactly. We will meet at <a href="https://goo.gl/maps/4kfABNV6Pc92" title="Tseng Tau Tsuen (井頭村)" target="_blank">Tseng Tau Tsuen (井頭村)</a>, walk a few minutes to <a href="https://goo.gl/maps/R4qv67i4KX42" title="Kei Ling Ha Hoi Pier (企嶺下海碼頭), aka Sam Pui Chau Pier (三杯酒碼頭)" target="_blank">Kei Ling Ha Hoi Pier (企嶺下海碼頭), aka Sam Pui Chau Pier (三杯酒碼頭)</a>, and take a boat to <a href="https://goo.gl/maps/P7zwgrFRq8t" title="Yung Shue O rafts (榕樹澳漁排)" target="_blank">Yung Shue O rafts (榕樹澳漁排)</a>, which is a group of fish cultivation facilities floating on the sea.</div>} />
                <FaqItem
                    question="What if there is bad weather?"
                    answer={
                        <div>
                            <p>
                                <strong>Typhoon</strong><br />
                                The activity will be re-scheduled when Typhoon Signal No. 8 or above is hoisted. Please check our <a href="https://9gag.slack.com/messages/G5L8FC3SQ">Slack channel</a> for announcements.
                            </p>
                            <p>
                                <strong>Rainstorm</strong><br />
                                If an Amber, Red, or Black Rainstorm Warning Signal is in force <i>before</i> the activity, we may decide to re-schedule the activity. Please check our <a href="https://9gag.slack.com/messages/G5L8FC3SQ">Slack channel</a> for announcements.<br />
                                If an Amber, Red, or Black Rainstorm Warning Signal is in force <i>during</i> the activity, the activity will continue.
                            </p>
                        </div>
                    } />
                <FaqItem
                    question="Can I swim there?"
                    answer="Sure! The water quality there is good. Swimming is one of the typical activities there. Remember to bring your swimwear." />
                /*
                <FaqItem
                    question={'What is Star beach? What\'s so special about it?'}
                    answer="Blah blah blah" />
                */
                <FaqItem
                    question="What types of fish can I catch there?"
                    answer="Usually in summer, you will find groups of baby snappers (鱲魚) and some mottled spinefoot (泥鯭). Sometimes if you are lucky, you may catch small groupers (石斑)." />
                <FaqItem
                    question="Is there any mobile network coverage?"
                    answer="Of course! It is covered by all mobile network providers." />
            </div>
        );
    }
}

export default muiThemeable()(Faq);
