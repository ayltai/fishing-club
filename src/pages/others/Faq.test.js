'use strict';

import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Faq from './Faq';

it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
            <MuiThemeProvider>
                <Faq />
            </MuiThemeProvider>
        </BrowserRouter>,
        div);

    expect(div.innerHTML).toMatchSnapshot();
});
