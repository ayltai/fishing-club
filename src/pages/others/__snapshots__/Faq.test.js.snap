// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `

<div data-reactroot>
  <div class="faqItem"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; z-index: 1;"
  >
    <div style="padding-bottom: 0px;">
      <div style="padding: 16px; font-weight: 500; box-sizing: border-box; position: relative; white-space: nowrap; cursor: pointer;">
        <div style="display: inline-block; vertical-align: top; white-space: normal; padding-right: 90px;">
          <span style="color: rgba(0, 0, 0, 0.87); display: block; font-size: 15px;">
            Are we going to rent a yacht and fishing at sea?
          </span>
          <span style="color: rgba(0, 0, 0, 0.54); display: block; font-size: 14px;">
          </span>
        </div>
        <button tabindex="0"
                type="button"
                style="border: 10px; box-sizing: border-box; display: inline-block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: auto; padding: 12px; outline: none; font-size: 0; font-weight: inherit; position: absolute; z-index: 1; overflow: visible; width: 48px; height: 48px; top: 0px; bottom: 0px; right: 4px; background: none;"
        >
          <div>
            <svg viewbox="0 0 24 24"
                 style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: currentColor; height: 24px; width: 24px;"
            >
              <path d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z">
              </path>
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div class="faqItem"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; z-index: 1;"
  >
    <div style="padding-bottom: 0px;">
      <div style="padding: 16px; font-weight: 500; box-sizing: border-box; position: relative; white-space: nowrap; cursor: pointer;">
        <div style="display: inline-block; vertical-align: top; white-space: normal; padding-right: 90px;">
          <span style="color: rgba(0, 0, 0, 0.87); display: block; font-size: 15px;">
            What if there is bad weather?
          </span>
          <span style="color: rgba(0, 0, 0, 0.54); display: block; font-size: 14px;">
          </span>
        </div>
        <button tabindex="0"
                type="button"
                style="border: 10px; box-sizing: border-box; display: inline-block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: auto; padding: 12px; outline: none; font-size: 0; font-weight: inherit; position: absolute; z-index: 1; overflow: visible; width: 48px; height: 48px; top: 0px; bottom: 0px; right: 4px; background: none;"
        >
          <div>
            <svg viewbox="0 0 24 24"
                 style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: currentColor; height: 24px; width: 24px;"
            >
              <path d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z">
              </path>
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div class="faqItem"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; z-index: 1;"
  >
    <div style="padding-bottom: 0px;">
      <div style="padding: 16px; font-weight: 500; box-sizing: border-box; position: relative; white-space: nowrap; cursor: pointer;">
        <div style="display: inline-block; vertical-align: top; white-space: normal; padding-right: 90px;">
          <span style="color: rgba(0, 0, 0, 0.87); display: block; font-size: 15px;">
            Can I swim there?
          </span>
          <span style="color: rgba(0, 0, 0, 0.54); display: block; font-size: 14px;">
          </span>
        </div>
        <button tabindex="0"
                type="button"
                style="border: 10px; box-sizing: border-box; display: inline-block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: auto; padding: 12px; outline: none; font-size: 0; font-weight: inherit; position: absolute; z-index: 1; overflow: visible; width: 48px; height: 48px; top: 0px; bottom: 0px; right: 4px; background: none;"
        >
          <div>
            <svg viewbox="0 0 24 24"
                 style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: currentColor; height: 24px; width: 24px;"
            >
              <path d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z">
              </path>
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div class="faqItem"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; z-index: 1;"
  >
    <div style="padding-bottom: 0px;">
      <div style="padding: 16px; font-weight: 500; box-sizing: border-box; position: relative; white-space: nowrap; cursor: pointer;">
        <div style="display: inline-block; vertical-align: top; white-space: normal; padding-right: 90px;">
          <span style="color: rgba(0, 0, 0, 0.87); display: block; font-size: 15px;">
            What is Star beach? What's so special about it?
          </span>
          <span style="color: rgba(0, 0, 0, 0.54); display: block; font-size: 14px;">
          </span>
        </div>
        <button tabindex="0"
                type="button"
                style="border: 10px; box-sizing: border-box; display: inline-block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: auto; padding: 12px; outline: none; font-size: 0; font-weight: inherit; position: absolute; z-index: 1; overflow: visible; width: 48px; height: 48px; top: 0px; bottom: 0px; right: 4px; background: none;"
        >
          <div>
            <svg viewbox="0 0 24 24"
                 style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: currentColor; height: 24px; width: 24px;"
            >
              <path d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z">
              </path>
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div class="faqItem"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 2px; z-index: 1;"
  >
    <div style="padding-bottom: 0px;">
      <div style="padding: 16px; font-weight: 500; box-sizing: border-box; position: relative; white-space: nowrap; cursor: pointer;">
        <div style="display: inline-block; vertical-align: top; white-space: normal; padding-right: 90px;">
          <span style="color: rgba(0, 0, 0, 0.87); display: block; font-size: 15px;">
            Is there any mobile network coverage?
          </span>
          <span style="color: rgba(0, 0, 0, 0.54); display: block; font-size: 14px;">
          </span>
        </div>
        <button tabindex="0"
                type="button"
                style="border: 10px; box-sizing: border-box; display: inline-block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: auto; padding: 12px; outline: none; font-size: 0; font-weight: inherit; position: absolute; z-index: 1; overflow: visible; width: 48px; height: 48px; top: 0px; bottom: 0px; right: 4px; background: none;"
        >
          <div>
            <svg viewbox="0 0 24 24"
                 style="display: inline-block; color: rgba(0, 0, 0, 0.87); fill: currentColor; height: 24px; width: 24px;"
            >
              <path d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z">
              </path>
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>

`;
